<h2>Data Ssientist Job Market Analysis In India</h2>

In [2]:
import requests
import bs4
from bs4 import BeautifulSoup

In [3]:
from time import sleep

In [4]:
import pandas as pd
import numpy as np

In [24]:

indeed_cities = ['San+Francisco', 'California', 'San+Diego', 'Sunnyvale' \
                'Mountain+View', 'San+Jose', 'Los+Angeles', 'Palo+Alto', \
                'Redwood+City', 'Menlo+Park', 'Santa+Clara', 'South+San+Francisco', \
                'El+Segundo', 'Santa+Clara+Valley', 'Irvine']

In [25]:
max_results_per_city = 2000

results = []



for city in indeed_cities:
    for start in range(0, max_results_per_city, 50):
        #for .in extraction
        #url = "https://www.indeed.co.in/jobs?q=Data+Scientist&l=" + city + "&limit=50&radius=25&start=" + str(start)
        #for .com extraction
        url = "https://www.indeed.com/jobs?q=Data+Scientist&l=" + city + "&limit=50&radius=25&start=" + str(start)
        print(url)
        html = requests.get(url)
        soup = BeautifulSoup(html.text, 'html.parser')
        for result in soup.find_all('div', {'class':' row result'}):
            results.append(result)
        sleep(1)
print(url_count)

https://www.indeed.com/jobs?q=Data+Scientist&l=San+Francisco&limit=50&radius=25&start=0
https://www.indeed.com/jobs?q=Data+Scientist&l=San+Francisco&limit=50&radius=25&start=50
https://www.indeed.com/jobs?q=Data+Scientist&l=San+Francisco&limit=50&radius=25&start=100
https://www.indeed.com/jobs?q=Data+Scientist&l=San+Francisco&limit=50&radius=25&start=150
https://www.indeed.com/jobs?q=Data+Scientist&l=San+Francisco&limit=50&radius=25&start=200
https://www.indeed.com/jobs?q=Data+Scientist&l=San+Francisco&limit=50&radius=25&start=250
https://www.indeed.com/jobs?q=Data+Scientist&l=San+Francisco&limit=50&radius=25&start=300
https://www.indeed.com/jobs?q=Data+Scientist&l=San+Francisco&limit=50&radius=25&start=350
https://www.indeed.com/jobs?q=Data+Scientist&l=San+Francisco&limit=50&radius=25&start=400
https://www.indeed.com/jobs?q=Data+Scientist&l=San+Francisco&limit=50&radius=25&start=450
https://www.indeed.com/jobs?q=Data+Scientist&l=San+Francisco&limit=50&radius=25&start=500
https://www.i

https://www.indeed.com/jobs?q=Data+Scientist&l=San+Diego&limit=50&radius=25&start=650
https://www.indeed.com/jobs?q=Data+Scientist&l=San+Diego&limit=50&radius=25&start=700
https://www.indeed.com/jobs?q=Data+Scientist&l=San+Diego&limit=50&radius=25&start=750
https://www.indeed.com/jobs?q=Data+Scientist&l=San+Diego&limit=50&radius=25&start=800
https://www.indeed.com/jobs?q=Data+Scientist&l=San+Diego&limit=50&radius=25&start=850
https://www.indeed.com/jobs?q=Data+Scientist&l=San+Diego&limit=50&radius=25&start=900
https://www.indeed.com/jobs?q=Data+Scientist&l=San+Diego&limit=50&radius=25&start=950
https://www.indeed.com/jobs?q=Data+Scientist&l=San+Diego&limit=50&radius=25&start=1000
https://www.indeed.com/jobs?q=Data+Scientist&l=San+Diego&limit=50&radius=25&start=1050
https://www.indeed.com/jobs?q=Data+Scientist&l=San+Diego&limit=50&radius=25&start=1100
https://www.indeed.com/jobs?q=Data+Scientist&l=San+Diego&limit=50&radius=25&start=1150
https://www.indeed.com/jobs?q=Data+Scientist&l=San

https://www.indeed.com/jobs?q=Data+Scientist&l=San+Jose&limit=50&radius=25&start=1150
https://www.indeed.com/jobs?q=Data+Scientist&l=San+Jose&limit=50&radius=25&start=1200
https://www.indeed.com/jobs?q=Data+Scientist&l=San+Jose&limit=50&radius=25&start=1250
https://www.indeed.com/jobs?q=Data+Scientist&l=San+Jose&limit=50&radius=25&start=1300
https://www.indeed.com/jobs?q=Data+Scientist&l=San+Jose&limit=50&radius=25&start=1350
https://www.indeed.com/jobs?q=Data+Scientist&l=San+Jose&limit=50&radius=25&start=1400
https://www.indeed.com/jobs?q=Data+Scientist&l=San+Jose&limit=50&radius=25&start=1450
https://www.indeed.com/jobs?q=Data+Scientist&l=San+Jose&limit=50&radius=25&start=1500
https://www.indeed.com/jobs?q=Data+Scientist&l=San+Jose&limit=50&radius=25&start=1550
https://www.indeed.com/jobs?q=Data+Scientist&l=San+Jose&limit=50&radius=25&start=1600
https://www.indeed.com/jobs?q=Data+Scientist&l=San+Jose&limit=50&radius=25&start=1650
https://www.indeed.com/jobs?q=Data+Scientist&l=San+Jos

In [39]:
def get_loc(result):
    try:
        return result.find('span', {'class': 'location'}).text
    except:
        return np.nan

In [40]:
def get_comp(result):
    try:
        return result.find('span', {'class': 'company'}).text
    except:
        return np.nan

In [41]:
def get_job(result):
    try:
        return result.find('a', {'data-tn-element': 'jobTitle' }).text
    except:
        return np.nan

In [42]:
def get_sal(result):
    try:
        return result.find('span', {'class': 'no-wrap'}).text
    except:
        return np.nan

In [43]:
def get_desc(result):
    try:
        return result.find('span', {'class': 'summary'}).text
    except:
        return np.nan

In [44]:
jobs0 = pd.DataFrame(columns=['location', 'title', 'company', 'salary', 'summary'])

In [45]:
for entry in results:
    location = get_loc(entry)
    title = get_job(entry)
    #print(title)
    company = get_comp(entry)
    #print(company)
    salary = get_sal(entry)
    desc = get_desc(entry)
    #print(jobs0.loc[len(jobs0)])
    
    jobs0.loc[len(jobs0)] = [location, title, company, salary, desc]
jobs0.head()

location  \
0  San Francisco, CA 94111 (Financial District area)   
1                                  San Francisco, CA   
2     San Francisco, CA 94103 (South Of Market area)   
3                                  San Francisco, CA   
4                                  San Francisco, CA   

                                              title                 company  \
0                                    Data Scientist  \n\n    BAIN & COMPANY   
1                Data Scientist - San Francisco, CA         \n    SymbaSync   
2  Machine Learning Engineer (Connections), AI Labs            \n\n    Uber   
3                     Data Analyst / Data Scientist   \n    Need To Impeach   
4                                    Data Scientist          \n    King.com   

                                         salary  \
0                                           NaN   
1  \n                $105,000 - $135,000 a year   
2                                           NaN   
3                                           NaN   
4                                           NaN   

                                             summary  
0  \n            Experience with explaining and a...  
1  \n            You will build machine learning ...  
2  \n            You will work closely with resea...  
3  \n            Need To Impeach is seeking a Dat...  
4  \n            We are looking for a talented an...

In [46]:
#replacing the \n
jobs0 = jobs0.replace('\n', '', regex=True)

In [47]:
jobs0.head()

location  \
0  San Francisco, CA 94111 (Financial District area)   
1                                  San Francisco, CA   
2     San Francisco, CA 94103 (South Of Market area)   
3                                  San Francisco, CA   
4                                  San Francisco, CA   

                                              title              company  \
0                                    Data Scientist       BAIN & COMPANY   
1                Data Scientist - San Francisco, CA            SymbaSync   
2  Machine Learning Engineer (Connections), AI Labs                 Uber   
3                     Data Analyst / Data Scientist      Need To Impeach   
4                                    Data Scientist             King.com   

                                       salary  \
0                                         NaN   
1                  $105,000 - $135,000 a year   
2                                         NaN   
3                                         NaN   
4                                         NaN   

                                             summary  
0              Experience with explaining and app...  
1              You will build machine learning mo...  
2              You will work closely with researc...  
3              Need To Impeach is seeking a Data ...  
4              We are looking for a talented and ...

In [48]:
jobs0.shape

(23299, 5)

In [49]:
jobs0.isnull().sum()

location        0
title           0
company         0
salary      22180
summary         0
dtype: int64

Majority of the companies doesen't' declare the salaries. So we will be considering either Mean or Median Salary for the Analysis

<b>Converting object into CSV file</b>

In [51]:
jobs0.to_csv('datascientist-job-results.csv', index=False, encoding='utf-8')

In [5]:
#Reading CSV File
df_indeed_result = pd.read_csv('datascientist-job-results.csv')

In [6]:
#Deleting all the non_matching titles from the dataset

#Key words to match the titles
title_keywords = ['data', 'Machine', 'Scientist', 'Analyst', 'SAS', 'Statistician', \
                    'Analytics', 'Science']
#dictionary to store all matching titles
title_dictionary = []

#isolating title from dataset
title = df_indeed_result['title']

for job_title in title:
    #print(job_title)
    for title_split in job_title.split():
        #print(title_split)
        if title_split in title_keywords:
            title_dictionary.append(job_title)
title_dictionary[:5]

['Data Scientist',
 'Data Scientist - San Francisco, CA',
 'Machine Learning Engineer (Connections), AI Labs',
 'Data Analyst / Data Scientist',
 'Data Analyst / Data Scientist']

In [7]:
#filtering the matching titles with dictionary
df_results_title_filter = df_indeed_result[df_indeed_result.title.apply(lambda x: x in title_dictionary)]
print(df_results_title_filter.shape)

#display only matching role for datascientist
df_results_title_filter.head()

(11119, 5)


location  \
0  San Francisco, CA 94111 (Financial District area)   
1                                  San Francisco, CA   
2     San Francisco, CA 94103 (South Of Market area)   
3                                  San Francisco, CA   
4                                  San Francisco, CA   

                                              title              company  \
0                                    Data Scientist       BAIN & COMPANY   
1                Data Scientist - San Francisco, CA            SymbaSync   
2  Machine Learning Engineer (Connections), AI Labs                 Uber   
3                     Data Analyst / Data Scientist      Need To Impeach   
4                                    Data Scientist             King.com   

                                       salary  \
0                                         NaN   
1                  $105,000 - $135,000 a year   
2                                         NaN   
3                                         NaN   
4                                         NaN   

                                             summary  
0              Experience with explaining and app...  
1              You will build machine learning mo...  
2              You will work closely with researc...  
3              Need To Impeach is seeking a Data ...  
4              We are looking for a talented and ...

In [8]:
#checking for the duplicates
df_results_duplicates = df_results_title_filter[df_results_title_filter.duplicated()]
df_results_duplicates.sort_values('company', inplace=True)
df_results_duplicates[:5]

/Users/afnity_computer/anaconda3/lib/python3.5/site-packages/ipykernel_launcher.py:3: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy
  This is separate from the ipykernel package so we can avoid doing imports until


location  \
16326  Mountain View, CA 94041   
11176  Mountain View, CA 94041   
7637   Mountain View, CA 94041   
7638   Mountain View, CA 94041   
17097  Mountain View, CA 94041   

                                                   title      company salary  \
16326                    Scientist - Population Genetics      23andMe    NaN   
11176  Scientist / Senior Scientist - Statistical Gen...      23andMe    NaN   
7637           Scientist / Senior Scientist - Health R&D      23andMe    NaN   
7638   Scientist / Senior Scientist - Computational B...      23andMe    NaN   
17097  Scientist / Senior Scientist - Statistical Gen...      23andMe    NaN   

                                                 summary  
16326              We are seeking a team member who i...  
11176              For senior scientists:. Supervise,...  
7637               Communicate project results both v...  
7638               For senior scientists:. Supervise,...  
17097              For senior scientists:. Supervise,...

In [60]:
#Dropping duplicates
df_result_final = df_results_title_filter.drop_duplicates(subset=['company', 'title', 'location', 'summary'])
df_result_final.shape
df_result_final.to_csv("indeed_ds_job_data.csv", index=False, encoding='utf-8')

In [69]:
#Top 50 Companies listing out jobs

from collections import Counter
company_job_list = Counter(df_result_final.company)

Total_jobs_company = pd.DataFrame.from_dict(company_job_list,  orient='index').reset_index()
Total_jobs_company = Total_jobs_company.rename(columns={'index': 'company', 0:'job_count'})

Total_jobs_company['job_count'] = Total_jobs_company['job_count'].astype('int')
Total_jobs_company['company'] = Total_jobs_company['company'].astype('str')

print(Total_jobs_company.dtypes)

#sorting dataframe
df_job_count = Total_jobs_company.sort_values(by='job_count', axis=0, ascending=False).reset_index(drop = True)
top_listing = df_job_count.head(10)


#Converting into csv file
top_listing.to_csv("ds_top50_job_listing.csv", index=False)

top_listing[:10]

company      object
job_count     int64
dtype: object


company  job_count
0                            Apple         88
1                       Amazon.com         51
2                         Facebook         50
3                           Google         42
4                            Adobe         30
5      Cedars-Sinai Medical Center         27
6                           JD.com         26
7                          Harnham         23
8        The Aerospace Corporation         18
9               Jobspring Partners         18

<div>
    <a href="https://plot.ly/~srini_hubble/15/?share_key=BoWSwcp9YK0VPTFedZjH9t" target="_blank" title="top10_job_listing_companies" style="display: block; text-align: center;"><img src="https://plot.ly/~srini_hubble/15.png?share_key=BoWSwcp9YK0VPTFedZjH9t" alt="top10_job_listing_companies" style="max-width: 100%;width: 700px;"  width="700" onerror="this.onerror=null;this.src='https://plot.ly/404.png';" /></a>
    <script data-plotly="srini_hubble:15" sharekey-plotly="BoWSwcp9YK0VPTFedZjH9t" src="https://plot.ly/embed.js" async></script>
</div>


In [64]:
#The role description for top 10 companies
#df_role_description = df_result_final[top_listing['company']]
top_10_listing = top_listing['company']
top_10_listing = list(top_10_listing[:10])


#filtering the matching titles with dictionary
top_10_company_role_description = df_result_final[df_result_final.company.apply(lambda x: x in top_10_listing)]

top_10_company_role_description = top_10_company_role_description[['company', 'summary']]
top_10_company_role_description[:10]

top_10_company_role_description.to_csv("top10_ds_description.csv", index=False)

top_10_company_role_description[:10]

company                                            summary
8          Facebook              The Data Scientist Analytics role ...
17            Apple              We are looking for people with exc...
74           Google              Deep interest and aptitude in data...
161           Adobe              We’re looking for machine learning...
167          Google              Participate in cutting edge resear...
205      Amazon.com              As a Machine Learning Engineer you...
217          Google              Manage projects and deliverables, ...
230      Amazon.com              We are seeking a Senior Applied Sc...
233          Google              Deep interest and aptitude in data...
250           Adobe              Apply data mining and machine lear...

<b>Trending Job Titles for Data Scientists</b>

In [65]:
#Top 10 titles under the datascintist indeed category

company_job_title = Counter(df_result_final.title)

Top_titles = pd.DataFrame.from_dict(company_job_title,  orient='index').reset_index()
Top_titles = Top_titles.rename(columns={'index': 'title', 0:'count_titles'})

Top_titles['count_titles'] = Top_titles['count_titles'].astype('int')
Top_titles['title'] = Top_titles['title'].astype('str')

print(Top_titles.dtypes)

#sorting dataframe
top_10_job_titles = Top_titles.sort_values(by='count_titles', axis=0, ascending=False).reset_index(drop = True)
top_10_job_titles = top_10_job_titles[:10]

top_10_job_titles.to_csv("top10_job_titles.csv", index=False)
top_10_job_titles

title           object
count_titles     int64
dtype: object


title  count_titles
0                       Data Scientist           146
1                Senior Data Scientist            67
2            Machine Learning Engineer            38
3                   Research Scientist            26
4                     Research Analyst            22
5             Principal Data Scientist            16
6                   Sr. Data Scientist            16
7                         Data Analyst            15
8                  Lead Data Scientist            14
9  Software Engineer, Machine Learning            12

<div>
    <a href="https://plot.ly/~srini_hubble/20/?share_key=lXOLfM5i5AGH8ZBe0e0zDp" target="_blank" title="Trending_titles_datascience" style="display: block; text-align: center;"><img src="https://plot.ly/~srini_hubble/20.png?share_key=lXOLfM5i5AGH8ZBe0e0zDp" alt="Trending_titles_datascience" style="max-width: 100%;width: 600px;"  width="600" onerror="this.onerror=null;this.src='https://plot.ly/404.png';" /></a>
    <script data-plotly="srini_hubble:20" sharekey-plotly="lXOLfM5i5AGH8ZBe0e0zDp" src="https://plot.ly/embed.js" async></script>
</div>


<h3>Total Jobs by City</h3>

In [70]:
#Leading City in USA for Data Science Jobs
#Top 10 titles under the datascintist indeed category

company_job_location = Counter(df_result_final.location)

Top_cities = pd.DataFrame.from_dict(company_job_location,  orient='index').reset_index()
Top_cities = Top_cities.rename(columns={'index': 'city', 0:'count_jobs'})

Top_cities['count_jobs'] = Top_cities['count_jobs'].astype('int')
Top_cities['city'] = Top_cities['city'].astype('str')

print(Top_cities.dtypes)

#sorting dataframe
job_dist_by_city = Top_cities.sort_values(by='count_jobs', axis=0, ascending=False).reset_index(drop = True)
job_dist_by_city = job_dist_by_city.head(10)

job_dist_by_city.to_csv("city_vs_jobs.csv", index=False)
job_dist_by_city[:10]

city          object
count_jobs     int64
dtype: object


city  count_jobs
0       San Francisco, CA         282
1           San Diego, CA         111
2         Los Angeles, CA          99
3            San Jose, CA          99
4           Sunnyvale, CA          91
5  Santa Clara Valley, CA          85
6       Mountain View, CA          81
7           Palo Alto, CA          69
8          Menlo Park, CA          65
9         Santa Clara, CA          40

<div>
    <a href="https://plot.ly/~srini_hubble/25/?share_key=n8UsKEXfwBpFpKcVScdD34" target="_blank" title="top_10_cities_dataScience_jobs" style="display: block; text-align: center;"><img src="https://plot.ly/~srini_hubble/25.png?share_key=n8UsKEXfwBpFpKcVScdD34" alt="top_10_cities_dataScience_jobs" style="max-width: 100%;width: 600px;"  width="600" onerror="this.onerror=null;this.src='https://plot.ly/404.png';" /></a>
    <script data-plotly="srini_hubble:25" sharekey-plotly="n8UsKEXfwBpFpKcVScdD34" src="https://plot.ly/embed.js" async></script>
</div>

<h3>Salary Information</h3>

The data for Salary is highly messed, so it's a littile tricky task to clean this data. For this Analysis we are using the salry information of for Year and incase of range converted into average salary

In [28]:
salaries = df_result_final[df_result_final.salary.notnull()]

In [29]:
salaries.shape

(138, 5)

In [30]:
salaries.head()

location                                title  \
1         San Francisco, CA   Data Scientist - San Francisco, CA   
51          Foster City, CA  Data Scientist / Sr. Data Scientist   
52  South San Francisco, CA                       Data Scientist   
54        San Francisco, CA  Data Scientist (Big Data Analytics)   
55        San Francisco, CA           Data Scientist (Analytics)   

                    company  \
1                 SymbaSync   
51                     Visa   
52      Outpost Games, Inc.   
54                   Ahrefs   
55                 Opendoor   

                                               salary  \
1                          $105,000 - $135,000 a year   
51                  $98,000 - $125,000 a year (Ind...   
52                  $103,000 - $131,000 a year (In...   
54                  $124,000 - $158,000 a year (In...   
55                  $115,000 - $147,000 a year (In...   

                                              summary  
1               You will build machine learning mo...  
51              To support our rapidly growing gro...  
52              Solid understanding of statistical...  
54              Good knowledge in a wide variety o...  
55              As a Data Scientist you will:. Cra...

In [31]:

df_cleaned_salaries = salaries[(~salaries.salary.str.contains('an hour')) & (~salaries.salary.str.contains('a month'))
                   & (~salaries.salary.str.contains('a week')) & (~salaries.salary.str.contains('a day'))]


df_cleaned_salaries.salary = salaries.salary.str.replace('a year', \
                                                         '').str.replace(',', \
                                                                         '').str.replace('$', '').str.replace('(Indeed est.)',\
                                                                                                              '')

/Users/afnity_computer/anaconda3/lib/python3.5/site-packages/pandas/core/generic.py:3643: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy
  self[name] = value


In [45]:
#Variation in the input and output salaries
#Its because we deleted hours and monthly based salaries from dataset
df_cleaned_salaries.shape

(124, 5)

<b>Removing extra paranthesis in salary variable</b>

In [46]:
salary_data_cleaned = []

for x in df_cleaned_salaries.salary:
    #removing extra paranthesis
    x = x.split('()')
    if len(x) > 1:
        del x[1]
    salary_data_cleaned.append(x)
        
print(salary_data_cleaned[:2])  
print(len(salary_data_cleaned))

[['                105000 - 135000 '], ['                98000 - 125000  ']]
124


<b>Turn salary ranges to an average and convert that into floats</b>

In [47]:
#nested for loop to handle multi dimensional list
#salary_data_cleaned is multi dimensional

new_salaries = []
for salary_clean in salary_data_cleaned:
    for i in salary_clean:
        a = i.split('-')
        if len(a) == 2:
            new_salaries.append(np.mean([float(b) for b in a]))
        else:
            new_salaries.append(float(a[0]))

In [48]:
new_salaries[0:10]

[120000.0,
 111500.0,
 117000.0,
 141000.0,
 131000.0,
 129500.0,
 125000.0,
 122500.0,
 155500.0,
 143000.0]

In [49]:
df_cleaned_salaries.salary = new_salaries

/Users/afnity_computer/anaconda3/lib/python3.5/site-packages/pandas/core/generic.py:3643: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy
  self[name] = value


In [50]:
df_cleaned_salaries.shape

(124, 5)

In [51]:
df_cleaned_salaries[:5]

location                                title  \
1         San Francisco, CA   Data Scientist - San Francisco, CA   
51          Foster City, CA  Data Scientist / Sr. Data Scientist   
52  South San Francisco, CA                       Data Scientist   
54        San Francisco, CA  Data Scientist (Big Data Analytics)   
55        San Francisco, CA           Data Scientist (Analytics)   

                    company    salary  \
1                 SymbaSync  120000.0   
51                     Visa  111500.0   
52      Outpost Games, Inc.  117000.0   
54                   Ahrefs  141000.0   
55                 Opendoor  131000.0   

                                              summary  
1               You will build machine learning mo...  
51              To support our rapidly growing gro...  
52              Solid understanding of statistical...  
54              Good knowledge in a wide variety o...  
55              As a Data Scientist you will:. Cra...

<b>Univariate Analysis</b>

In [52]:
company_salary = round(df_cleaned_salaries['salary'], 1)

<b>Median Salary</b>

In [53]:
company_salary.median()

139250.0

The Median Salary of DataScientis is $139,250 for annum

<b>Mean Salary</b>

In [54]:
round(company_salary.mean(), 1)

134296.5

The Median Salary of DataScientis is $134,296 for annum

<b>Average Salary by Cities</b>

In [56]:
salary_city = df_cleaned_salaries[['location', 'salary']]
salary_city[:5]

location    salary
1         San Francisco, CA  120000.0
51          Foster City, CA  111500.0
52  South San Francisco, CA  117000.0
54        San Francisco, CA  141000.0
55        San Francisco, CA  131000.0

In [71]:
salary_city_mean = salary_city.groupby(['location'])['salary'].mean()
salary_city_mean = round(salary_city_mean, 1)
#round(salary_city_mean['salary'], 1)
salary_city_mean.to_csv("avg_salary_by_city.csv")
salary_city_mean[:50]

location
Berkeley, CA                                         102525.0
Burbank, CA 91505                                     70000.0
Cypress, CA                                          140000.0
Edwards AFB, CA                                       72993.0
Foster City, CA                                      111500.0
Irvine, CA                                           120000.0
Los Angeles, CA                                      134197.6
Los Angeles, CA 90001                                140000.0
Menlo Park, CA                                        78582.0
Menlo Park, CA 94026                                  94000.0
Mountain View, CA                                    172500.0
Palo Alto, CA                                        186666.7
Pasadena, CA                                          60000.0
Redondo Beach, CA                                    108500.0
Redwood City, CA                                     165000.0
Redwood Shores, CA                                   135500.0

<div>
    <a href="https://plot.ly/~srini_hubble/32/?share_key=aa8Kdj5yeOKzElVzMZDJ7m" target="_blank" title="Avg Salary by City" style="display: block; text-align: center;"><img src="https://plot.ly/~srini_hubble/32.png?share_key=aa8Kdj5yeOKzElVzMZDJ7m" alt="Avg Salary by City" style="max-width: 100%;width: 800px;"  width="800" onerror="this.onerror=null;this.src='https://plot.ly/404.png';" /></a>
    <script data-plotly="srini_hubble:32" sharekey-plotly="aa8Kdj5yeOKzElVzMZDJ7m" src="https://plot.ly/embed.js" async></script>
</div>


<h3>Conclusion</h3>

1. Apple is the Tob Job listing company for the Data Scientist Roles
2. SanFrancisco CA is the better option to look out for a Data Scientist
3. Data Scientist is the trending title, probably most of these roles for Beginner level or 2-3    years experienc
4. Median Salary for the DataScientist is 139,250 Dollars
5. Mean Salary for the DataScientist is 134,296 Dollars
6. Avg Salary in San Mateo, CA  is 170,000 Dollars and it is the maximum
7. Sherman Oaks, CA 91423 has the lowest Avg salary of $40000